skglm/tests/test_datafits.py

```diff
@@ -114,5 +115,34 @@ def test_gamma():
     np.testing.assert_allclose(clf.coef_, gamma_results.params, rtol=1e-6)
 
 
+def test_cox():
+    rng = np.random.RandomState(1265)
+    n_samples, n_features = 10, 30
```

Comment:
This test basically tests the shapes of the returned argument but the maths could be wrong. could we use automatic diff in this test to check that we have the correct gradient and upper bound on Hessian?

Reply:
+1 for autodiff. Not sure about `check_grad`, isn't it based on finite difference?

Reply:
@mathurinm, @PABannier, any idea about how to unittest the upper bound on the Hessian?

The explicit expression of the Hessian is huge (considering also Efron)